client/src/components/Search/Search.js

```diff
@@ -267,6 +269,30 @@ class Search extends Component {
     setTimeout(() => this.focusFirstFormField(selector), 50);
   }
 
+  onClickOut(evt) {
+    // This is a fairly rudimentory way to detect if the user clicked on an entirely different
+    // DOM element, which probably uses jQuery
+    // It's important that we only call show() when a relevent element is clicked, otherwise we'll
+    // end up making lots of unnecessary AJAX requests to the form schema endpoint
+    const el = evt.target;
+    let doShow = false;
+    // Match on target ID
+    ['filters-button'].forEach(id => {
+      if (el.id === id) {
+        doShow = true;
+      }
+    });
+    // Match on target class
+    ['font-icon-search', 'filter-open'].forEach(className => {
+      if (el.classList.contains(className)) {
+        doShow = true;
+      }
+    });
+    if (doShow) {
```

Comment:
```suggestion
    if (el.matches('#filters-button, .font-icon-search, .filter-open')) {
```
https://developer.mozilla.org/en-US/docs/Web/API/Element/matches

Reply:
Updated